server/src/main/java/org/apache/druid/server/coordination/SegmentLoadDropHandler.java

```diff
@@ -63,7 +68,9 @@ public class SegmentLoadDropHandler implements DataSegmentChangeHandler
   private final SegmentLoaderConfig config;
   private final DataSegmentAnnouncer announcer;
   private final SegmentManager segmentManager;
-  private final ScheduledExecutorService exec;
+  private final ScheduledExecutorService scheduledExecutorService;
```

Comment:
I don't think we really need a third thread pool, you can just make the normal thread pool a `ScheduledThreadPoolExecutor` and always use it for drops.

Reply:
Changed